lms/templates/dashboard/_dashboard_entitlement_actions.html

```diff
@@ -0,0 +1,49 @@
+<%page args="course_overview, entitlement, dashboard_index" expression_filter="h"/>
+
+<%!
+from django.utils.translation import ugettext as _
+from django.core.urlresolvers import reverse
+%>
+
+<div class="wrapper-entitlement-action-more">
+
+  ## NOTE: In order for the dropdown functionality provided by lms/static/js/dashboard/dropdown.js to work,
+  ##       this button must include 'id', 'data-dropdown-id', 'data-dropdown-button-id',
+  ##       and class='entitlement-action-more'.
+  <button id="entitlement-actions-dropdown-link-${dashboard_index}"
+          class="entitlement-action entitlement-action-more"
+          type="button"
+          aria-haspopup="true"
+          aria-expanded="false"
+          aria-controls="entitlement-actions-dropdown-${dashboard_index}"
+          data-dropdown-id="#entitlement-actions-dropdown-${dashboard_index}"
+          data-dropdown-button-id="#entitlement-actions-dropdown-link-${dashboard_index}">
+    <span class="sr">${_('Course options for {courseName}').format(courseName=course_overview.display_name_with_default)}</span>
+    <span class="fa fa-cog" aria-hidden="true"></span>
+  </button>
+
+  <div id="entitlement-actions-dropdown-${dashboard_index}" class="entitlement-actions-dropdown" tabindex="-1">
+    <ul id="entitlement-actions-dropdown-list-${dashboard_index}" class="entitlement-actions-dropdown-list" aria-label="${_('Available Actions')}" role="menu">
+      <li id="entitlement-actions-item-unenroll-${dashboard_index}" class="entitlement-actions-item" role="menuitem">
+
+        ## NOTE: In order for the modal functionality provided by lms/static/js/leanModal.js to work,
+        ##       'href' must be set to be the 'id' of the modal that should be opened on click,
+        ##       'rel' must be set to 'leanModal', and data-model-close-button must be set to
+        ##       '.close-entitlement-modal'. Also, in order to properly toggle the gear dropdown on
+        ##       click, this link will need to define 'data-dropdown-id' and 'data-dropdown-button-id'.
+        ##       Finally, class='js-action-unenroll-entitlement' needs to be set so that the
+        ##       EntitlementUnenrollView can find this link and properly attach click handlers.
+        <a href="#unenroll-entitlement-modal" id="action-unenroll-entitlement-${dashboard_index}" class="entitlement-action js-action-unenroll-entitlement" rel="leanModal"
+          data-modal-close-button=".close-entitlement-modal"
+          data-dropdown-id="#entitlement-actions-dropdown-${dashboard_index}"
+          data-dropdown-button-id="#entitlement-actions-dropdown-link-${dashboard_index}"
+          data-course-name="${course_overview.display_name_with_default}"
+          data-course-number="${course_overview.number}"
+          data-entitlement-is-refundable="${entitlement.is_entitlement_refundable()}"
+          data-entitlement-api-endpoint="${reverse('entitlements_api:v1:enrollments', args=[unicode(entitlement.uuid)])}">
```

Comment:
It's a bit worrying to pass through all of this through data objects. A user could just go into the DOM, change the value, and then trick our JS into thinking an entitlement is refundable that is not. 

We should have logic on the backend to handle this, but it's still risky and if you can find a way to avoid this that would be a safer bet here.

Reply:
As mentioned above, the only data attribute that actually matters for the request is `api-endpoint`. Everything else is only used for display/modal/dropdown logic. The server is ultimately responsible for determining whether or not an entitlement is refundable. We cannot and do not specify whether or not a user should receive a refund for an entitlement via the API.